lib/internal/util/types.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const ReflectApply = Reflect.apply;
+
+// This function is borrowed from the function with the same name on V8 Extras'
+// `utils` object. V8 implements Reflect.apply very efficiently in conjunction
+// with the spread syntax, such that no additional special case is needed for
+// function calls w/o arguments.
+// Refs: https://github.com/v8/v8/blob/d6ead37d265d7215cf9c5f768f279e21bd170212/src/js/prologue.js#L152-L156
+function uncurryThis(func) {
+  return (thisArg, ...args) => ReflectApply(func, thisArg, args);
+}
+
+function getterFunc(obj, prop) {
+  return Object.getOwnPropertyDescriptor(obj, prop).get;
+}
+
+const TypedArrayPrototype = Object.getPrototypeOf(Uint8Array.prototype);
+
+const TypedArrayProto_toStringTag =
+    uncurryThis(getterFunc(TypedArrayPrototype, Symbol.toStringTag));
```

Comment:
This took me some time to understand. Maybe just a comment would help, or inline the `getterFunc` to make it obvious what is being done.

Reply:
Okay.